src/hooks/FreelyTransferable.sol

```diff
@@ -0,0 +1,171 @@
+import {BitmapLib} from "src/misc/libraries/BitmapLib.sol";
+
+import {IRoot} from "src/common/interfaces/IRoot.sol";
+import {ITransferHook, HookData, ESCROW_HOOK_ID} from "src/common/interfaces/ITransferHook.sol";
+
+import {IShareToken} from "src/spoke/interfaces/IShareToken.sol";
+
+import {IFreezable} from "src/hooks/interfaces/IFreezable.sol";
+import {IMemberlist} from "src/hooks/interfaces/IMemberlist.sol";
+import {UpdateRestrictionType, UpdateRestrictionMessageLib} from "src/hooks/libraries/UpdateRestrictionMessageLib.sol";
+
+/// @title  Freely Transferable
+/// @notice Hook implementation that:
+///         * Allows any non-frozen account to receive tokens and transfer tokens
+///         * Requires accounts to be added as a member before submitting a deposit or redemption request
+///         * Supports freezing accounts which blocks transfers both to and from them
+///
+/// @dev    The first 8 bytes (uint64) of hookData is used for the memberlist valid until date,
+///         the last bit is used to denote whether the account is frozen.
+contract FreelyTransferable is Auth, IMemberlist, IFreezable, ITransferHook {
+    using BitmapLib for *;
+    using UpdateRestrictionMessageLib for *;
+    using BytesLib for bytes;
+    using CastLib for bytes32;
+
+    /// @dev Least significant bit
+    uint8 public constant FREEZE_BIT = 0;
+
+    IRoot public immutable root;
+
+    constructor(address root_, address deployer) Auth(deployer) {
+        root = IRoot(root_);
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Callback from share token
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc ITransferHook
+    function onERC20Transfer(address from, address to, uint256 value, HookData calldata hookData)
+        external
+        virtual
+        returns (bytes4)
+    {
+        require(checkERC20Transfer(from, to, value, hookData), TransferBlocked());
+        return ITransferHook.onERC20Transfer.selector;
+    }
+
+    /// @inheritdoc ITransferHook
+    function onERC20AuthTransfer(
+        address, /* sender */
+        address, /* from */
+        address, /* to */
+        uint256, /* value */
+        HookData calldata /* hookData */
+    ) external pure returns (bytes4) {
+        return ITransferHook.onERC20AuthTransfer.selector;
+    }
+
+    /// @inheritdoc ITransferHook
+    function checkERC20Transfer(address from, address to, uint256, /* value */ HookData calldata hookData)
+        public
+        view
+        returns (bool)
+    {
+        uint128 fromHookData = uint128(hookData.from);
+        if (fromHookData.getBit(FREEZE_BIT) == true && !root.endorsed(from)) {
+            // Source is frozen and not endorsed
+            return false;
+        }
+
+        uint128 toHookData = uint128(hookData.to);
+        if (toHookData.getBit(FREEZE_BIT) == true) {
```

Comment:
QUESTION 

Should we have this here? IIRC, we said we don't need that from a legal point of view?

Reply:
I think having freeze feature is always good optionality